language/solana/move-to-solana/src/stackless/extensions.rs

```diff
@@ -21,19 +21,101 @@ pub impl<'a> ModuleEnvExt for mm::ModuleEnv<'a> {
 #[extension_trait]
 pub impl<'a> FunctionEnvExt for mm::FunctionEnv<'a> {
     fn llvm_symbol_name(&self, tyvec: &[mty::Type]) -> String {
-        let mut name = self.get_full_name_str();
+        let name = self.get_full_name_str();
         if name == "<SELF>::<SELF>" {
             // fixme move-model names script fns "<SELF>".
             // we might want to preserve the actual names
             "main".to_string()
         } else {
-            for ty in tyvec {
-                name += &format!("_{}", ty.display(&self.get_type_display_ctx()))
-            }
-            name.replace([':', '<', '>'], "_").replace(", ", "_")
```

Comment:
In this case, the first call to replace is replacing any of several single-char patterns with a string. The second case is replace a multi-char string with a string.

Doing both can't be done in one call to replace because the first parameters are two different (array of char vs. string).

This could probably be written with the input just as an array of strings:

```
name.replace([":", "<", ">", ", "], "_")
```

It's slightly less efficient because searching for the patterns then requires iterating over strings instead of single chars.

Note this snippet of code is being removed in this patch.